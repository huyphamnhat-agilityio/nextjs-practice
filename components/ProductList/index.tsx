"use client";
// Constants
import { HOME_LIMIT, INITIAL_PAGE } from "@/constants";

// Services
import { getProducts } from "@/lib";

// Components
import { ProductCard } from "../common";

// Types
import { Pagination, Product } from "@/types";

const ProductList = ({
  // currentPage = INITIAL_PAGE,
  // limit = HOME_LIMIT,
  // query = "",
  products,
}: {
  // currentPage: number;
  // limit?: number;
  // query?: string;
  products?: Pagination<Product>;
}) => {
  // const products = await getProducts({ page: currentPage, limit, query });

  return (
    <div className="grid grid-col-1 lg:grid-cols-2 xl:grid-cols-3 gap-2.5 justify-evenly">
      {products?.data.map((product) => (
        <ProductCard key={product.id} {...product} />
      ))}
    </div>
  );
};

export default ProductList;
