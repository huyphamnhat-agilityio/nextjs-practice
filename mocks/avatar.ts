import { Avatar } from "@/types";

export const MOCK_AVATARS: Array<Avatar> = [
  {
    src: "/feedback-avatar-1.jpg",
    alt: "An avatar of a user",
  },
  {
    src: "/feedback-avatar-2.jpg",
    alt: "An avatar of a user",
  },
  {
    src: "/feedback-avatar-3.jpg",
    alt: "An avatar of a user",
  },
];
