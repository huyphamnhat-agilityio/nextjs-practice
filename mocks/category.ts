import { Category } from "@/types";

export const MOCK_CATEGORIES: Array<Category> = [
  { id: "1", name: "Training Courses" },
  { id: "2", name: "Books Library" },
  { id: "3", name: "Certified Teachers" },
];
