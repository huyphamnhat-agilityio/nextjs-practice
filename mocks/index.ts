export * from "./category";
export * from "./product";
export * from "./fetch";
export * from "./avatar";
export * from "./image";
