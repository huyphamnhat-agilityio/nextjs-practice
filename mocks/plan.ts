import { PlanCardProps } from "@/components/common/PlanCard";

export const MOCK_PLAN: PlanCardProps = {
  isNew: false,
  title: "free",
  subTitle: "Organize across all apps by hand",
  price: 19,
  duration: "month",
  description: "Slate helps you see how many more days you need...",
  features: [
    {
      isSupported: true,
      feature: "Unlimited product updates",
    },
    {
      isSupported: true,
      feature: "Unlimited product updates",
    },
    {
      isSupported: true,
      feature: "Unlimited product updates",
    },
    {
      isSupported: false,
      feature: "1GB Cloud storage",
    },
    {
      isSupported: false,
      feature: "Email and community support",
    },
  ],
};
