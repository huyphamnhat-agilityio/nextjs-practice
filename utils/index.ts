export * from "./test-utils";
export * from "./format";
export * from "./convert";
