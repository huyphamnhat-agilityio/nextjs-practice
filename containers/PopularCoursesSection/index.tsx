// Components
import { Pagination, ProductCard } from "@/components";

// Constants
import { LIMIT } from "@/constants";

// Services
import { getProducts } from "@/services";

const PopularCoursesSection = async ({
  currentPage,
}: {
  currentPage: number;
}) => {
  const products = await getProducts(currentPage, LIMIT);
  return (
    <section>
      <div className="bg-dark-blue mt-32.5">
        <div className="max-w-lg md:max-w-4xl lg:max-w-5xl my-0 mx-auto py-40 flex flex-col gap-20">
          <div className="flex flex-col gap-2.5 text-center 2xl:text-start">
            <h6 className="text-primary text-sm/6 font-bold">
              Practice Advice
            </h6>
            <h2 className="text-white text-5xl/12.5 font-bold">
              Our Popular Courses
            </h2>
            <p className="text-white text-sm">
              Problems trying to resolve the conflict between
              <br />
              the two major realms of Classical physics: Newtonian mechanics
            </p>
          </div>
          <div className="flex flex-wrap gap-10 2xl:gap-2.5 justify-evenly">
            {products.data.map((product) => (
              <ProductCard key={product.id} {...product} />
            ))}
          </div>

          <Pagination total={products.totalPages} initialPage={products.page} />
        </div>
      </div>
    </section>
  );
};

export default PopularCoursesSection;
