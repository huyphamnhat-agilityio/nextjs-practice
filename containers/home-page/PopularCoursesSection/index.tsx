import { Suspense } from "react";

// Components
import { Pagination, ProductList, ProductListSkeleton } from "@/components";

// Constants
import { LIMIT } from "@/constants";

// Services
import { getProducts } from "@/lib";
import { notFound } from "next/navigation";

const PopularCoursesSection = async ({
  currentPage,
}: {
  currentPage: number;
}) => {
  const products = await getProducts(currentPage, LIMIT);

  // if (!products) {
  //   notFound();
  // }

  return (
    <section>
      <div className="bg-dark-blue mt-32.5 flex justify-center">
        <div className="max-w-xs sm:max-w-lg md:max-w-xl lg:max-w-4xl xl:max-w-5xl my-0 py-40 flex flex-col gap-20">
          <div className="flex flex-col gap-2.5 text-center 2xl:text-start">
            <p className="text-primary text-sm/6 font-bold">Practice Advice</p>
            <h2 className="text-white text-5xl/12.5 font-bold">
              Our Popular Courses
            </h2>
            <p className="text-white text-sm">
              Problems trying to resolve the conflict between
              <br />
              the two major realms of Classical physics: Newtonian mechanics
            </p>
          </div>

          <Suspense key={currentPage} fallback={<ProductListSkeleton />}>
            <ProductList currentPage={currentPage} />
          </Suspense>

          <Pagination total={products.totalPages} initialPage={products.page} />
        </div>
      </div>
    </section>
  );
};

export default PopularCoursesSection;
