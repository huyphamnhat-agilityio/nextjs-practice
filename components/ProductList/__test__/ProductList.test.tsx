import { render } from "@testing-library/react";

// Components
import PopularCoursesSection from "..";

// Utils
import { resolvedComponent } from "@/utils";

// Mocks
import { MOCK_PRODUCTS, mockFetch } from "@/mocks";

// Components
import ProductList from "..";

describe("ProductList test cases", () => {
  it("should render correctly", async () => {
    window.fetch = mockFetch({ data: MOCK_PRODUCTS });

    const ProductListResolved = await resolvedComponent<{
      currentPage: number;
    }>(ProductList, {
      currentPage: 1,
    });

    const { container } = render(<ProductListResolved />);

    expect(container).toMatchSnapshot();
  });
});
