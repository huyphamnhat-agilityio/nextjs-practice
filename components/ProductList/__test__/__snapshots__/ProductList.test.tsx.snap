// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList test cases should render correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-col-1 lg:grid-cols-2 xl:grid-cols-3 gap-2.5 justify-evenly"
  />
</DocumentFragment>
`;
